File: cpp/base/src/main/vsdk/toolkit/environment/material/ShadingType.cpp
#include "vsdk/toolkit/environment/material/ShadingType.h"
ShadingType ShadingTypeUtil::fromCode(int code)
{
    switch ( code ) {
        case SHADING_NOLIGHT: return SHADING_NOLIGHT;
        case SHADING_FLAT: return SHADING_FLAT;
        case SHADING_GOURAUD: return SHADING_GOURAUD;
        case SHADING_PHONG: return SHADING_PHONG;
        case SHADING_COOK_TERRANCE: return SHADING_COOK_TERRANCE;
        default: return SHADING_GOURAUD;
    }
}

ShadingType ShadingTypeUtil::next(ShadingType value)
{
    return static_cast<ShadingType>((static_cast<int>(value) + 1) % 5);
}

File: typescript/base/src/main/vsdk/toolkit/environment/material/ShadingType.ts
export enum ShadingType {
    NOLIGHT = 0,
    FLAT = 1,
    GOURAUD = 2,
    PHONG = 3,
    COOK_TERRANCE = 4,
}
export namespace ShadingType {
    export const fromCode = (x: number): ShadingType => (x >= 0 && x <= 4 ? (x as ShadingType) : ShadingType.GOURAUD);
    export const next = (x: ShadingType): ShadingType => ((x + 1) % 5) as ShadingType;
}
